src/typescript-service.ts

```diff
@@ -338,6 +338,847 @@ export default class TypeScriptService {
 		});
 	}
 
+	getDocumentSymbol(params: DocumentSymbolParams): Promise<SymbolInformation[]> {
+		const uri = params.textDocument.uri;
+		return this.ensureFilesForHoverAndDefinition(uri).then(() => {
+			const fileName = util.uri2path(uri);
+			const config = this.projectManager.getConfiguration(uri);
+			return config.prepare(this.connection).then(() => {
+				this.projectManager.syncConfiguration(config, this.connection);
```

Comment:
If I omit `syncConfiguration`, I get no results back. To test this, remove this line and run `npm test`.